crates/ruff_cli/src/cache.rs

```diff
@@ -19,33 +20,44 @@ use ruff_text_size::{TextRange, TextSize};
 
 use crate::diagnostics::Diagnostics;
 
-/// On disk representation of a cache of a package.
-#[derive(Deserialize, Debug, Serialize)]
-pub(crate) struct PackageCache {
+/// [`Path`] that is relative to the package root in [`PackageCache`].
+pub(crate) type RelativePath = Path;
+/// [`PathBuf`] that is relative to the package root in [`PackageCache`].
+pub(crate) type RelativePathBuf = PathBuf;
+
+/// Cache.
+///
+/// `Cache` holds everything required to display the diagnostics for a single
+/// package. The on-disk representation is represented in `PackageCache` (and
+/// related) types.
+///
+/// This type manages the cache file, reading it from disk and writing it back
+/// to disk (if required).
+pub(crate) struct Cache {
     /// Location of the cache.
-    ///
-    /// Not stored on disk, just used as a storage location.
-    #[serde(skip)]
     path: PathBuf,
-    /// Path to the root of the package.
+    /// Package cache read from disk.
+    package: PackageCache,
+    /// Changes made compared to the (old) `package`.
     ///
-    /// Usually this is a directory, but it can also be a single file in case of
-    /// single file "packages", e.g. scripts.
-    package_root: PathBuf,
-    /// Mapping of source file path to it's cached data.
-    // TODO: look into concurrent hashmap or similar instead of a mutex.
-    files: Mutex<HashMap<RelativePathBuf, FileCache>>,
+    /// Recreated [`PackageCache::files`] which only contains the cache for
+    /// source file that still exist.
```

Comment:
```suggestion
    /// source files that still exist.
```

What's the meaning of *still exist*? That have not been deleted or that have been seen?